calicoctl/resourcemgr/resourcemgr.go

```diff
@@ -505,3 +518,103 @@ func (rh resourceHelper) GetTableTemplate(headings []string) (string, error) {
 
 	return buf.String(), nil
 }
+
+// validateMetadata compares the Metadata for a ResourceObject and warns if anything was
+// changed that will not be changed as a part of the update. Any discrepancies to the metadata
+// will be changed to match the metadata available in the stored object except for Labels and
+// Annotations.
+func validateMetadata(stored, changed ResourceObject) ResourceObject {
+	sm := stored.GetObjectMeta()
+	cm := changed.GetObjectMeta()
+
+	// Set the fields that are allowed to be overwritten (Labels and Annotations)
+	// so that they will not be overwritten.
+	// TODO: Figure out if a warning is needed
+	sm.SetAnnotations(cm.GetAnnotations())
+	sm.SetLabels(cm.GetLabels())
+
+	// Make a copy of the resource so we do not need to modify the input
+	resOut := changed.DeepCopyObject().(ResourceObject)
+	sm.(*v1.ObjectMeta).DeepCopyInto(resOut.GetObjectMeta().(*v1.ObjectMeta))
+	return resOut
+	// TODO: get rid of this if we do not need the individualized field warnings.
+	/*
+		if sm.GetNamespace() != cm.GetNamespace() {
+			log.Warning("Cannot change Metadata.Namespace. Reverting to past value.")
+			cm.SetNamespace(sm.GetNamespace())
+		}
+		if sm.GetName() != cm.GetName() {
+			log.Warning("Cannot change Metadata.Name. Reverting to past value.")
+			cm.SetName(sm.GetName())
+		}
+		if sm.GetGenerateName() != cm.GetGenerateName() {
```

Comment:
Personally I'd keep things simple and not bother warning about fields they can't change.  I'd probably just add some comments to the docs to list which fields can and cannot be modified through calicoctl.

Reply:
Fair enough. I'll get rid of the warnings.